Add react-strict-dom/postcss-plugin based on @stylexjs/postcss-plugin

javascripter opened the pull request:
# Why
https://github.com/facebook/react-strict-dom/issues/281#issuecomment-2712591645

This PR upstreams [postcss-react-strict-dom](https://github.com/javascripter/postcss-react-strict-dom) to the official repo to make it easier for maintainers to make changes.

I added RSD compatibility to `@stylexjs/postcss-plugin` in [this PR](https://github.com/facebook/stylex/pull/1063), so this adds `react-strict-dom/postcss-plugin` that re-exports the StyleX PostCSS Plugin with the required configuration for RSD consumers.

# How
- Added PostCSS Plugin to `packages/react-strict-dom/postcss/plugin.js`
- Export the plugin as `react-strict-dom/postcss-plugin` in package.json
- Updated `apps/examples` to use the official plugin and removed postcss-react-strict-dom dependency
- Updated docs to use the official plugin
- Updated StyleX to 0.14

# Test Plan
- [x] Run `cd apps/example && npm run dev:web` both from the original repo and this branch and compare side-by-side to make sure output matches 

Left is this branch, right is original
<img width="1376" alt="Screenshot 2025-05-20 at 15 12 39" src="https://github.com/user-attachments/assets/fba7670f-d58c-4fd6-a681-d60af25d51d6" />

necolas commented on packages/react-strict-dom/postcss/plugin.js:
```diff
@@ -0,0 +1,41 @@
+}) => {
+  include = [
+    // Include the React Strict DOM package's source files by default
+    require.resolve('react-strict-dom'),
```
I think we need to include the runtime export too

javascripter replied:
Added in this commit
https://github.com/facebook/react-strict-dom/pull/308/commits/e760f8c94d4bfe39a8f96a237caf4179313c0c1d

necolas commented on packages/react-strict-dom/postcss/plugin.js:
```diff
@@ -0,0 +1,41 @@
+    babelConfig,
+    include,
+    exclude,
+    useCSSLayers: false,
```
```suggestion
    useCSSLayers: true,
```

javascripter replied:
It seems Expo Web doesn't support CSS Layers. If we enable useCSS Layers and run `npm run dev:web`, it shows the below warning.
I disabled CSS layers for now:
https://github.com/facebook/react-strict-dom/pull/308/commits/b455d438e11581e55d709635e53b27ad5d74740a
```
| Warning: Unknown at rule: @property (src/app/stylex.css:16:10):
|   14 |
|   15 |  @layer priority1{
| > 16 | @property --backgroundColor { syntax: "*"; inherits: false;}
|      |          ^
|   17 | @property --opacity { syntax: "*"; inherits: false;}
|   18 | @property --transform { syntax: "*"; inherits: false;}
|   19 | :root, .xxw3bu5{--primary100-x1ud21w4:black;--primary200-x19lv8z8:white;}
| Warning: Unknown at rule: @property (src/app/stylex.css:17:10):
|   15 |  @layer priority1{
|   16 | @property --backgroundColor { syntax: "*"; inherits: false;}
| > 17 | @property --opacity { syntax: "*"; inherits: false;}
```

necolas replied:
Looks like it might be a StyleX bug. Don't think you're allowed to nest `@property` inside `@layer` in CSS

necolas commented:
#311 updates StyleX to 0.13

javascripter commented:
Rebased from main and updated to use `useCSSLayers: true`. Waiting for https://github.com/facebook/stylex/issues/1071 to land so CSS layer warnings will disapper

necolas commented:
This PR aims to avoid the Expo warnings https://github.com/facebook/stylex/pull/1087

necolas commented:
We should be able to proceed with this change now that the StyleX upgrade is done
